crates/amaru-ledger/src/governance.rs

```diff
@@ -1,34 +1,88 @@
 use crate::store::{columns::dreps::Row, Snapshot, StoreError};
-use amaru_kernel::{encode_bech32, DRep, StakeCredential};
+use amaru_kernel::{
+    encode_bech32, epoch_from_slot, Anchor, DRep, Epoch, StakeCredential, DREP_EXPIRY,
+    GOV_ACTION_LIFETIME,
+};
 use serde::ser::SerializeStruct;
-use std::collections::BTreeMap;
+use std::collections::{BTreeMap, HashSet};
 
 #[derive(Debug)]
-// TODO: Also add DRep information to this summary
 pub struct DRepsSummary {
     pub delegations: BTreeMap<StakeCredential, DRep>,
+    pub dreps: BTreeMap<DRep, DRepState>,
+}
+
+#[derive(Debug, serde::Serialize)]
+pub struct DRepState {
+    pub mandate: Epoch,
+    pub metadata: Option<Anchor>,
 }
 
 impl DRepsSummary {
     pub fn new(db: &impl Snapshot) -> Result<Self, StoreError> {
-        let dreps = db
+        let mut dreps = BTreeMap::new();
+
+        let current_epoch = db.epoch();
+        let mut all_proposals_epochs = db
+            .iter_proposals()?
+            .map(|(key, _)| epoch_from_slot(key.transaction_pointer.slot))
+            .collect::<HashSet<_>>();
+        // TODO filter out proposals that have been ratified
+
+        all_proposals_epochs.insert(163);
+        all_proposals_epochs.insert(165);
```

Comment:
This shall be unnecessary, provided that we correctly import proposals from snapshots. I'll make sure we do.